packages/orchestration/src/orchestration-client.ts

```diff
@@ -89,12 +129,40 @@ export class OrchestrationClient {
         },
         controller
       );
+      return streamResponse;
     } catch (error) {
       controller.abort();
       throw error;
     }
   }
 
+  getMessageHistory(): ChatMessages | undefined {
+    if (!this.useClientHistory) {
+      throw new Error(
+        'Message history is not enabled for this client. Please enable it by passing `useClientHistory: true` in the client configuration.'
+      );
+    }
+    return this.messagesHistory;
+  }
+
+  private async *processStreamEnd(
+    stream: OrchestrationStream<OrchestrationStreamChunkResponse>,
+    response?: OrchestrationStreamResponse<OrchestrationStreamChunkResponse>
+  ): AsyncGenerator<OrchestrationStreamChunkResponse> {
+    if (!response) {
+      throw new Error('Response is required to process stream end.');
+    }
+    for await (const chunk of stream) {
+      yield chunk;
+    }
+
+    response._openStream = false;
+
+    if (this.useClientHistory) {
+      this.messagesHistory = response.getAllMessages();
+    }
+  }
```

Comment:
[req] I understand why you add this here. I would still reuse `OrchestrationStream._processStreamEnd` and add another async generator in this file to add message history if there is no better way 🧐